turbopy/core.py

```diff
@@ -212,13 +212,15 @@ def read_clock_from_input(self):
     def read_tools_from_input(self):
         """Construct :class:`ComputeTools` based on input"""
         if "Tools" in self.input_data:
-            for tool_names, params in self.input_data["Tools"].items():
-                tool_class = ComputeTool.lookup(tool_names[0])
-                params["type"] = tool_names[0]
-                if "custom_name" in params:
-                    params["custom_name"] = tool_names[1]
-                self.compute_tools.append(tool_class(
-                    owner=self, input_data=params))
+            for tool_name, params in self.input_data["Tools"].items():
+                tool_class = ComputeTool.lookup(tool_name)
+                if len(params) >= 1:
```

Comment:
This won't work. Dictionaries have length, too. You need to check if it's a list:
```python
if type(params) is not list:
    params = [params]
```

Reply:
That length was for seeing whether I needed to submit the items inside to the compute_tools list or whether I could just submit params.

Reply:
I know. But `len(params) >= 1` can be true if `params` is a dictionary and not a list, and I still want to be able to just have it be a dictionary (if there is only one of any particular tool).

Reply:
Updated